product/views.py

```diff
@@ -1,3 +1,34 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http  import JsonResponse,HttpResponse
+from django.views import View
+
+from .models      import Product, Detail, Series, Media, Size
+
+class DetailView(View):
+
+    def get(self,request,product_code):
+        try:
+            selected_product = Product.objects.select_related('detail').prefetch_related('media_set', 'series_set').get(code = product_code)
+
+            data = [{
+                'code'         : selected_product.code,
+                'name'         : selected_product.name,
+                'price'        : selected_product.price,
+                'gender'       : selected_product.gender,
+                'summary'      : selected_product.detail.summary,
+                'color_name'   : selected_product.color_name,
+                'size_list'    : [selected_size.size for selected_size in Size.objects.prefetch_related('productsize_set').filter(product = selected_product)],
```

Comment:
Size로부터 productsize를 역참조 하지 않고 12라인에서 productsize도 prefetch로 역참조 후에 다음과 같이 size정보를 리스트로 만들어서 쿼리수를 줄여볼 수 있을 것 같습니다.
```suggestion
                'size_list'    : [size.size.size for size in selected_product.productsize_set.all()],
```